nx/lib/nx.ex

```diff
@@ -3265,6 +3265,38 @@ defmodule Nx do
     end)
   end
 
+  @doc ~S"""
+  Split tensor into train and test subsets.
+
+  ## Examples
+
+    Split into 80% for training and 20% for testing.
+
+    iex> train = Nx.tensor([[3, 6, 5],[26, 75, 3]])
+    iex> test = Nx.tensor([[23, 4, 1]])
+    iex> {^train, ^test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), split: 0.8)
```

Comment:
I think we should have an easier-to-read test here. Maybe allocate an iota with shape {10} for the first example, and a second example with {10, 2} and axis: 0 and another example with {2, 10} and axis: 1.

This would make it easier for people to read the example and see how things should behave in the nice cases.

And then show what happens for the not-nice case in a single example. Currently we only have the not-nice case as an example